docs/extending/development-environment.md

```diff
@@ -0,0 +1,192 @@
+
+While not strictly necessary for running the extension, these extensions are recommended for development:
+
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) - Integrates ESLint into VS Code.
+- [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) - Integrates Prettier into VS Code.
+
+The full list of recommended extensions is [here](https://github.com/Kilo-Org/kilocode/blob/main/.vscode/extensions.json)
+
+### Project Structure
+
+The project is organized into several key directories:
+
+- **`src/`** - Core extension code
+    - **`core/`** - Core functionality and tools
+    - **`services/`** - Service implementations
+- **`webview-ui/`** - Frontend UI code
+- **`e2e/`** - End-to-end tests
+- **`scripts/`** - Utility scripts
+- **`assets/`** - Static assets like images and icons
+
+## Development Workflow
+
+### Building the Extension
+
+To build the extension:
+
+```bash
+npm run build
+```
+
+This will:
+
+1. Build the webview UI
+2. Compile TypeScript
+3. Bundle the extension
+4. Create a `.vsix` file in the `bin/` directory
+
+### Running the Extension
+
+To run the extension in development mode:
+
+1. Press `F5` (or select **Run** → **Start Debugging**) in VSCode
+2. This will open a new VSCode window with Kilo Code loaded
+
+### Hot Reloading
+
+- **Webview UI changes**: Changes to the webview UI will appear immediately without restarting
+- **Core extension changes**: Changes to the core extension code will automatically reload the ext host
+
+In development mode (NODE_ENV="development"), changing the core code will trigger a `workbench.action.reloadWindow` command, so it is no longer necessary to manually start/stop the debugger and tasks.
+
+> **Important**: In production builds, when making changes to the core extension, you need to:
+>
+> 1. Stop the debugging process
+> 2. Kill any npm tasks running in the background (see screenshot below)
+> 3. Start debugging again
+
+<img width="600" alt="Stopping background tasks" src="https://github.com/user-attachments/assets/466fb76e-664d-4066-a3f2-0df4d57dd9a4" />
+
+### Installing the Built Extension
+
+To install your built extension:
+
+```bash
+code --install-extension "$(ls -1v bin/kilo-code-*.vsix | tail -n1)"
+```
+
+Replace `[version]` with the current version number.
+
+## Testing
+
+Kilo Code uses several types of tests to ensure quality:
+
+### Unit Tests
+
+Run unit tests with:
+
+```bash
+npm test
+```
+
+This runs both extension and webview tests.
+
+To run specific test suites:
+
+```bash
+npm run test:extension  # Run only extension tests
+npm run test:webview    # Run only webview tests
+```
+
+### End-to-End Tests
+
+E2E tests verify the extension works correctly within VSCode:
+
+1. Create a `.env.local` file in the root with required API keys:
+
+    ```
+    OPENROUTER_API_KEY=sk-or-v1-...
```

Comment:
Is this needed for us?

Reply:
Well, that's what https://github.com/Kilo-Org/kilocode/blob/main/e2e/VSCODE_INTEGRATION_TESTS.md says.